### PR TITLE
Provisioning + Quickstart samples moved to sdk repo

### PR DESCRIPTION
**Pull request by patilsnr**

### CODE REVIEW

**Review by brycewang-microsoft — COMMENTED** [review]
Seems like some Quickstarts samples are put under the hub **device** folder in SDK repo, like "[device-streams-echo](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Quickstarts/device-streams-echo)" and "[device-streams-proxy](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Quickstarts/device-streams-proxy)". However, those involves both device and service clients, so we may need to separate them in either device or service folder of the SDK repo, and add links to each other in the readme file.

File: iothub/device/samples/Getting Started/InvokeDeviceMethod/Program.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        {
+            Console.WriteLine("IoT Hub Quickstarts - InvokeDeviceMethod application.");
+
+            // Parse sample parameters.
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                .WithParsed(parsedParams => parameters = parsedParams)
+                .WithNotParsed(errors => Environment.Exit(1));
+
+            // This sample accepts the service connection string as a parameter, if present.
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/InvokeDeviceMethod/Program.cs:31]: As an environment variable?

File: iothub/device/samples/Getting Started/InvokeDeviceMethod/Program.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            Console.WriteLine($"Invoking direct method for device: {deviceId}");
+
+            // Invoke the direct method asynchronously and get the response from the simulated device.
+            CloudToDeviceMethodResult response = await s_serviceClient.InvokeDeviceMethodAsync(deviceId, methodInvocation);
+
+            Console.WriteLine($"Response status: {response.Status}, payload:\n\t{response.GetPayloadAsJson()}");
+
+        }
+
+        private static void ValidateConnectionString(string hubConnectionString)
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/InvokeDeviceMethod/Program.cs:63]: Consider moving this to the parameters class.

File: iothub/device/samples/Getting Started/InvokeDeviceMethod/Program.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+            // This sample accepts the service connection string as a parameter, if present.
+            ValidateConnectionString(parameters.HubConnectionString);
+
+            // Create a ServiceClient to communicate with service-facing endpoint on your hub.
+            s_serviceClient = ServiceClient.CreateFromConnectionString(parameters.HubConnectionString);
+
+            await InvokeMethodAsync(parameters.DeviceId);
+
+            s_serviceClient.Dispose();
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/InvokeDeviceMethod/Program.cs:39]: Move variable to method instance and use `using` keyword instead.

File: iothub/device/samples/Getting Started/SimulatedDevice/Program.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+            // SendDeviceToCloudMessagesAsync is designed to run until cancellation has been explicitly requested by Console.CancelKeyPress.
+            // As a result, by the time the control reaches the call to close the device client, the cancellation token source would
+            // have already had cancellation requested.
+            // Hence, if you want to pass a cancellation token to any subsequent calls, a new token needs to be generated.
+            // For device client APIs, you can also call them without a cancellation token, which will set a default
+            // cancellation timeout of 4 minutes: https://github.com/Azure/azure-iot-sdk-csharp/blob/64f6e9f24371bc40ab3ec7a8b8accbfb537f0fe1/iothub/device/src/InternalClient.cs#L1922
+            await s_deviceClient.CloseAsync();
+
+            s_deviceClient.Dispose();
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/SimulatedDevice/Program.cs:60]: Same as above. We can make this a local variable and use the `using` keyword instead.

File: iothub/device/samples/Getting Started/device-streams-echo/device/DeviceClientStreamingSample.csproj [hunk 5]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/device-streams-echo/device/DeviceClientStreamingSample.csproj:5]: net6.0?

File: iothub/device/samples/Getting Started/device-streams-echo/device/DeviceClientStreamingSample.csproj [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  </PropertyGroup>
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/device-streams-echo/device/DeviceClientStreamingSample.csproj:6]: lang 9.0?

File: iothub/device/samples/Getting Started/device-streams-echo/device/DeviceStreamSample.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+        private DeviceClient _deviceClient;
+
+        public DeviceStreamSample(DeviceClient deviceClient)
+        {
+            _deviceClient = deviceClient;
+        }
+
+        public async Task RunSampleAsync()
+        {
+            await RunSampleAsync(true).ConfigureAwait(false);
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/device-streams-echo/device/DeviceStreamSample.cs:24]: Remove `ConfigureAwait(false)` from samples.

File: iothub/device/samples/Getting Started/device-streams-echo/device/Properties/launchSettings.json.template [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{
+  "profiles": {
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/device-streams-echo/device/Properties/launchSettings.json.template:2]: remove this file

File: iothub/device/samples/Getting Started/ReadD2cMessages/Parameters.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+using CommandLine;
+
+namespace ReadD2cMessages
+{
+    /// <summary>
+    /// Parameters for the application.
+    /// </summary>
+    internal class Parameters
+    {
+        internal const string IotHubSharedAccessKeyName = "service";
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/ReadD2cMessages/Parameters.cs:13]: Can be moved to be a local method constant, yes?

File: iothub/device/samples/Getting Started/SimulatedDeviceWithCommand/Program.cs [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    {
+        private static DeviceClient s_deviceClient;
+        private static TimeSpan s_telemetryInterval = TimeSpan.FromSeconds(1); // Seconds
+
+        private static async Task Main(string[] args)
+        {
+            Parameters parameters = null;
+            ParserResult<Parameters> result = Parser.Default.ParseArguments<Parameters>(args)
+                    .WithParsed(parsedParams => parameters = parsedParams)
+                    .WithNotParsed(errors => Environment.Exit(1));
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/SimulatedDeviceWithCommand/Program.cs:33]: Tabbed in more than 4 spaces

File: iothub/device/samples/Getting Started/common/DeviceStreamingCommon.cs [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        /// <summary>
+        /// Creates a ClientWebSocket with the proper authorization header for Device Streaming.
+        /// </summary>
+        /// <param name="url">Url to the Streaming Gateway.</param>
+        /// <param name="authorizationToken">Authorization token to connect to the Streaming Gateway.</param>
+        /// <param name="cancellationToken">The token used for cancelling this operation if desired.</param>
+        /// <returns>A ClientWebSocket instance connected to the Device Streaming gateway, if successful.</returns>
+        public static async Task<ClientWebSocket> GetStreamingClientAsync(Uri url, string authorizationToken, CancellationToken cancellationToken)
+        {
+            ClientWebSocket wsClient = new ClientWebSocket();
```
Comment (drwill-ms) [inline iothub/device/samples/Getting Started/common/DeviceStreamingCommon.cs:22]:
```suggestion
            var wsClient = new ClientWebSocket();
```

**Review by drwill-ms — APPROVED** [review]

**Comment by brycewang-microsoft** [comment]
Seems like some samples in `/Quickstarts` and `/provisioning/Samples/service` is not included in this PR, is this expected?

**Comment by patilsnr** [comment]
> Seems like some samples in `/Quickstarts` and `/provisioning/Samples/service` is not included in this PR, is this expected?

The streaming samples from `Quickstarts` have been removed for this PR as we've decided to include them in a separate branch. The sample not included in `Provisioning\service` is news to me, however. It doesn't seem to show up in the solution explorer which is why it was missed. Going though it now.

**Review by brycewang-microsoft — APPROVED** [review]
Looks good!

**Pull request merged**